File: typescript/src/cost.ts
export class Cost {
    constructor(private rawCost: { cost: number }) {

    }

    public reduceByPercentage(percentOff: number) {
        return new Cost({cost: Math.ceil(this.rawCost.cost * (1 - percentOff / 100))})
    }

    getRaw() {
        return this.rawCost
    }

    static free() {
        return {cost: 0}
    }
}

File: python/src/services/cost.py
import math


class Cost:
    def __init__(self, cost: float):
        self.cost = cost

    def apply_percentage(self, reduction: int) -> None:
        self.cost = self.cost * (1 - reduction / 100)

    def round_up(self) -> int:
        return math.ceil(self.cost)
